### DISCUSSION TITLE
Support for IP Addresses

### DISCUSSION THREAD

**Discussion by obliqueRotationMethod** [adam-fowler/mqtt-nio#67] [Ideas]
Hey,

I'm using a Raspberry as MQTT Broker in my local network. I was trying to connect to it, get the error message:

> The operation couldn’t be completed. (NIOSSL.NIOSSLExtraError error 1.)
> MQTT Result is: failure(NIOSSLExtraError.cannotUseIPAddressInSNI: IP addresses cannot validly be used for Server Name Indication, got 192.168.178.75)

I would really like to use this lib for my project, any chance to add support for IP addresses?

Thanks and regards

Max

**Comment by adam-fowler** [comment]
The SSL certificates you are using for the MQTT server would have been created with a server name. If you initialize your `MQTTClient` with a `Configuration` that includes `sniServerName` set to the server name it should work.

**Reply by obliqueRotationMethod** [reply]
Great, that worked! Can connect to and receive messages from my broker.

Thank you very much! :)
